### PR TITLE
Feat: CORE-1269 - Install paddlepaddle wheel dependent on arch, supporting aarch64

### PR DESCRIPTION
**Pull request by tabossert**
Testing instructions

on Apple silicon

```
make docker-build
docker run -it unstructured:dev bash
python3
```
Then run the test in this PR https://unstructured-ai.atlassian.net/browse/CORE-1269

You should get output like shown in ticket

Run the same process on your local machine (not inside docker) with same test to verify the non aarch64 paddlepaddle got installed correctly

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,7 +22,8 @@ FROM base as deps
 # Copy and install Unstructured
 COPY requirements requirements
 
-RUN python3.10 -m pip install pip==${PIP_VERSION} && \
+RUN export ARCH=$(uname -m) && \
```
Comment (yuming-long) [inline Dockerfile:25]: should we move the docker file changes to base image since unstructured and unstructured  api images both need it?

Comment (tabossert) [reply Dockerfile:25]: Unfortunately no, cause it needs to be installed during building of our unstructured images as the base images are built on arm64 so our wheel would not get installed.  It's a little duplication between unstructured and unstructured-api but it should stay minimal

File: requirements/huggingface.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with Python 3.8
+# This file is autogenerated by pip-compile with Python 3.10
```
Comment (cragwolfe) [inline requirements/huggingface.txt:2]: please re-run `make pip-compile` with python3.8 (on mac)

**Comment by cragwolfe** [comment]
Should the extra also be added to https://github.com/Unstructured-IO/unstructured/blob/main/setup.py as with other extras?

Given that we don't yet have evidence showing evidence showing paddle outperforming tesseract (speed or quality), rather than bake this into the docker image, perhaps it is better that this is wrapped in an install script to make it easy for those that do want to experiment with it?

There has also been progress on resolving the need for the "paddleocr for tables only" pattern, lessening the urgency towards including paddleocr in all images: 

https://github.com/Unstructured-IO/unstructured-inference/pull/204
https://github.com/Unstructured-IO/unstructured-inference/pull/203

File: requirements/huggingface.txt (hunk 2 above)
Comment (yuming-long) [reply requirements/huggingface.txt:2]: actually i thought we move it to 3.10 (its the version from readme

**Comment by tabossert** [comment]
I already have a script with Makefile, I will push that and remove from Dockerfiles

File: scripts/install-paddleocr.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+# aarch64 requires a custom build of paddlepaddle
+if [ "${ARCH}" = "aarch64" ]; then
+		python3 -m pip install unstructured.paddlepaddle;
+else
+		python3 -m pip install paddlepaddle;
+fi
+python3 -m pip install paddleocr
```
Comment (yuming-long) [inline scripts/install-paddleocr.sh:9]: please change this to our forked version `unstructured_paddleocr`

File: setup.py [hunk 4]
```diff
@@ -153,6 +153,7 @@ def load_requirements(file_list: Optional[Union[str, List[str]]] = None) -> List
         # Legacy extra requirements
         "huggingface": load_requirements("requirements/huggingface.in"),
         "local-inference": all_doc_reqs,
+        "paddleocr": load_requirements("requirements/paddleocr.in"),
```
Comment (yuming-long) [inline setup.py:156]: we don't have a file named `paddleocr.in` right (rename extra-paddle.in to paddle.in ?)

**Comment by yuming-long** [comment]
actually, i think the paddle and paddleocr install setup should be moved to unstructured-inference repo

**Comment by cragwolfe** [comment]
> actually, i think the paddle and paddleocr install setup should be moved to unstructured-inference repo

strictly speaking, that does make more sense. however, i think it is OK to have it in `unstructured` for convenience.

**Comment by yuming-long** [comment]
> strictly speaking, that does make more sense. however, i think it is OK to have it in unstructured for convenience.

okay yeah so users can install it from unstructured library. but maybe not put it in `setup.py` ? i feel like inference repo also need to install paddle anyways. Or we can just keep it experimental and let user install it themself (so paddle not install fallback to tesseract?)

**Comment by cragwolfe** [comment]
> > strictly speaking, that does make more sense. however, i think it is OK to have it in unstructured for convenience.
> 
> okay yeah so users can install it from unstructured library. but maybe not put it in `setup.py` ? i feel like inference repo also need to install paddle anyways. Or we can just keep it experimental and let user install it themself (so paddle not install fallback to tesseract?)

right. i think having it in setup.py as an undocumented extra in setup.py is OK, but it's also fine with me to skip it for now. however, if paddle were in setup.py, it for sure should **not** be part of `all_doc_reqs`.

**Comment by yuming-long** [comment]
tested it works in docker container 🎉   can we add the pip install back into dockerfile

**Comment by cragwolfe** [comment]
> tested it works in docker container 🎉 can we add the pip install back into dockerfile

let's avoid pre-baking it into the image since it is not commonly used for now.

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,7 @@
 ### Enhancements
```
Comment (yuming-long) [inline CHANGELOG.md:3]: pls bump the change log version

File: requirements/extra-paddleocr.in [hunk 6]
```diff
@@ -0,0 +1,4 @@
+-c constraints.in
+-c base.txt
+
+unstructured.paddleocr==2.6.1.2
```
Comment (yuming-long) [inline requirements/extra-paddleocr.in:4]:
```suggestion
unstructured.paddleocr==2.6.1.2

```
new line nit

**Review by yuming-long — APPROVED** [review]
LGTM with some small nit suggestions

**Pull request merged**